src/Controller/CompanyController.php

```diff
@@ -11,68 +9,136 @@
      */
-    public function __construct(LoggerInterface $logger)
+    public function __construct(CompanyRepository $companyRepository)
     {
-        $this->logger = $logger;
+        $this->companyRepository = $companyRepository;
     }
-    
     /**
-     * loadTemplate
+     * adds Company to the db
      *
+     * @param  mixed $doctrine
      * @return Response
      */
-    public function loadTemplate():Response
+    public function addCompany(Request $request): Response
     {
-        return $this->render('company/index.html.twig', ['name'=>'Welcome']);
+        $name = $request->get('name');
+        $company = new Company();
+        $company->setName($name);
+        $companySaved = $this->companyRepository->save($company);
+
+        return new JsonResponse(
+            [
+             'saved' => $companySaved,
+            ]
+        );
     }
-    
     /**
-     * add fields for the form and the submitted data in varlog
+     * updates company name with id
      *
-     * @param  mixed $request
+     * @param  mixed $doctrine
+     * @param  mixed $id
      * @return Response
      */
-    public function add(Request $request):Response
+    public function updateCompany(Request $request, int $id): Response
     {
-        $this->form = $this->createFormBuilder()
-            ->add('name', TextType::class)
-            ->add('description', TextType::class)
-            ->add('email', EmailType::class)
-            ->add('saveData', SubmitType::class, ['label' => 'Create Company'])
-            ->getForm();
+        $requestParams = $request->query->all();
+        $updateResult = $this->companyRepository->update($id, $requestParams);
 
-        $this->form->handleRequest($request);
+        return new JsonResponse(
+            [
+                'rows_updated' => $updateResult
+            ]
+        );
+    }
+     /**
+     * deleteCompany
+     *
+     * @param  mixed $id
+     * @return Response
+     */
+    public function deleteCompany(int $id): Response
+    {
+        $companyForDelete = $this->companyRepository->find($id);
+        $companyDeleted = $this->companyRepository->remove($companyForDelete);
 
-        if ($this->form->isSubmitted() && $this->form->isValid()) {
-            $data=$this->form->getData();
-            $name=$data['name'];
-            $description= $data['description'];
-            $email= $data['email'];
+        return new JsonResponse(
+            [
+                'rows_deleted' => $companyDeleted
+            ]
+        );
+    }
+    /**
+     * listCompany
+     *
+     * @return Response
+     */
+    public function listCompany(): Response
+    {
+        $company = $this->companyRepository->listCompany();
 
+        return new JsonResponse(
+            [
+                'rows' => $company
+            ]
+        );
+    }
+    /**
+     * companyId
+     *
+     * @param  mixed $id
+     * @return Response
+     */
+    public function companyId(int $id): Response
```

Comment:
Not clear by name what this function does? Change it to something that is related to what the function does: ex: getCompanyById.

Check for all functions.